EVA-1447 Update README.md to test GitHub webhooks (#108)
EBIvariation/eva-ws: README.md

Build
-----

In order to build EVA, you need to install the Java Development Kit 8 and Maven.

All the dependencies will be downloaded by Maven.

> If you want to build all the modules, including dgva-server, you will need to [register with Oracle](https://login.oracle.com/mysso/signon.jsp) and [set up your maven security](https://docs.oracle.com/middleware/1213/core/MAVEN/config_maven_repo.htm#MAVEN9016) to be able to download from the Oracle Maven Repository the Oracle drivers used in dgva-server. See https://github.com/EBIvariation/eva-ws/issues/81#issuecomment-369212870 for more information.

Reviewer: Suggested to explain the motivation first, then how to achieve it:

```suggestion
If you want to build all the modules, including dgva-server, you will need to download the Oracle drivers used in dgva-server from the Oracle Maven Repository. In order to do so, [register with Oracle](https://login.oracle.com/mysso/signon.jsp) and [set up your Maven security](https://docs.oracle.com/middleware/1213/core/MAVEN/config_maven_repo.htm#MAVEN9016). See https://github.com/EBIvariation/eva-ws/issues/81#issuecomment-369212870 for more information.
```

                <eva.mongo.collections.annotations></eva.mongo.collections.annotations>
                <eva.mongo.collections.features></eva.mongo.collections.features>
                <eva.mongo.collections.files></eva.mongo.collections.files>
                <eva.mongo.collections.variants></eva.mongo.collections.variants>
            </properties>
        </profile>
    </profiles>

    <servers>
        <server>
            <id>maven.oracle.com</id>
            <username>your_oracle_user</username>
            <password>your_oracle_encrypted_password</password>
            <configuration>
              <basicAuthScope>
                <host>ANY</host>
                <port>ANY</port>
                <realm>OAM 11g</realm>
              </basicAuthScope>
              <httpConfiguration>
                <all>
                  <params>
                    <property>
                      <name>http.protocol.allow-circular-redirects</name>
                      <value>%b,true</value>
                    </property>
                  </params>
                </all>
              </httpConfiguration>
            </configuration>
      </server>
   </servers>
</settings>
```

> After setting up your maven security and maven settings, you can install the whole project and run the tests using the above "production" profile with `mvn clean install -Pproduction`. This will generate the ".war" files that you can deploy in Tomcat or other Java container. You have to set up the JNDI credentials in Tomcat (conf/context.xml), matching the "eva.evapro.datasource" and "eva.evapro.jdbc.url" properties.

Reviewer: ```suggestion
After setting up your Maven security and Maven settings, you can install the whole project and run the tests using the above "production" profile with `mvn clean install -Pproduction`. This will generate the ".war" files that you can deploy in Tomcat or other Java container. You have to set up the JNDI credentials in Tomcat (conf/context.xml), matching the "eva.evapro.datasource" and "eva.evapro.jdbc.url" properties.
```

> If you don't need to build dgva-server, you can do `mvn clean install -pl '!dgva-server'` to build the ".war" files of all the other modules only.

Reviewer: ```suggestion
If you don't need to build dgva-server, you can build the ".war" files for the rest of the modules running `mvn clean install -pl '!dgva-server'`.
```

Testing
-------

The tests implemented are both unit and integration tests. You can run them with `mvn test` from the root folder.

If you are compiling and want to skip the tests, you can do `mvn clean install -Pproduction -DskipTests`.

> For manual testing, you can deploy the ".war" files and go to the "swagger-ui.html" page to get an overview of the endpoints and run them manually.

Reviewer: We could show the explicit URL to swagger-ui.html in localhost to make the check a bit easier.
